### DISCUSSION TITLE
Is there a way to read zar file of this element?

### DISCUSSION THREAD

**Discussion by khoroshyy** [mjhoptics/ray-optics#158] [Q&A] [answered]
HI.
I am trying to use this element from Thorlabs.
https://www.thorlabs.com/thorproduct.cfm?partnumber=TTL200
but its model comes in zar file.
Is there any way to read it?
Thanks.

**Comment by mjhoptics** [comment]
Hello @khoroshyy,
Please look into the python package [zmxtools](https://github.com/tttom/zmxtools). I gather .zar is a zipped version of the zmx, etc. files. The tool takes a zar file as input and outputs a zmx file that ray-optics can (hopefully) read. I've used it in the past and seems to work.
Regards,
Mike Hayford

**Reply by khoroshyy** [reply]
Thanks. I am afraid we will be lost here. After communicating with the developer, zmxtools can unpack this type of zar. But it is a black-box lens, so it is my guess that Ray-Optics cannot work with it. 
Best regards. 
Petro,

**Reply by mjhoptics** [reply, marked as answer]
Ah, yes, no way around the encryption. Sometimes, zar files contain plain text data so the zmxtools package is good to have around.
Regards
Mike Hayford
